src/system_reader.rs

```diff
@@ -542,10 +630,39 @@ impl<R: RawReader> IonReader for SystemReader<R> {
                 self.lst.has_found_imports = false;
                 new_lst_state = NotReadingAnLst;
             }
-            ProcessingLstImports | ProcessingLstSymbols | ProcessingLstOpenContent => {
+            ProcessingLstSymbols | ProcessingLstOpenContent => {
+                // We're inside one of the LST fields. Finish processing the current level before
+                // stepping out.
+                self.finish_reading_current_level()?;
+                // If the upcoming call to step_out() will cause us to leave the field altogether,
+                // update our state to indicate that we're back at depth=1.
+                if self.depth() == 2 {
+                    new_lst_state = BetweenLstFields;
+                }
```

Comment:
This separate depth check would be replaced by adding new states for the fields within `imports`.

Reply:
`finish_reading_current_level()` doesn't change the LST import read states hence we need to explicitly change it using the depth check. Would you mind explaining your suggestion?

Reply:
> finish_reading_current_level() doesn't change the LST import read states

Doesn't it? It calls `next()`, which should cause the state to shift from `AtImportName` to `AtImportVersion` to `AtImportMaxId`, etc. Or am I missing something?